Automattic/wp-calypso
# Signup: Pre-fill domain name with site title

In the first step of the signup flow (`about`), the user can fill out their site title.
In the following step (`domain`), the user is asked to search and choose their site address.

Currently the domain search field doesn't have any initial value, even though almost always the site title and address are as similar as possible.

This PR pre-fills the domain search field with the site title the user inserted in the previous step.

## Testing Instructions

Start the signup flow from scratch (typically on `/start/about`).
Fill in the site title and proceed to next step.
The domain field should be populated with the site title, and there should already be some search results displayed.

## Notes

I had to add an `_isMounted` check to prevent firing the `onSearchChange` from `<Search>` when `<RegisterDomainStep>` wasn't fully mounted.
This happened because now, when the site title is set, we pass it as a default value to the search field, triggering that callback before its parent component is ready.

On a related note: pre-populating the `<Search>` input field throws a console warning:
> Failed prop type: You provided a `value` prop to a form field without an `onChange` handler. This will render a read-only field. If the field should be mutable use `defaultValue`. Otherwise, set either `onChange` or `readOnly`.

This is caused by the fact that `<Search>` uses `onInput` instead of `onChange` as a workaround for a [bug on IE11](https://github.com/Automattic/wp-calypso/blob/master/client/components/search/index.jsx#L367).

It might be easily fixable by replacing `value` with `defaultValue`, but despite being a one liner, there is a concrete risk of screwing up half Calypso.
